beacon-chain/p2p/gossipsub_peer_controller.go

```diff
@@ -0,0 +1,153 @@
+package p2p
+
+import (
+	"context"
+	"math"
+	"sync"
+	"time"
+
+	"github.com/OffchainLabs/prysm/v7/beacon-chain/p2p/gossipsubcrawler"
+	"github.com/OffchainLabs/prysm/v7/cmd/beacon-chain/flags"
+	"github.com/ethereum/go-ethereum/p2p/enode"
+	"github.com/libp2p/go-libp2p/core/peer"
+	"github.com/pkg/errors"
+)
+
+const (
+	peerPerTopic = 20
+)
+
+type GossipsubPeerDialer struct {
```

Comment:
I'm seeing all these types etc prefixed with "gossipsub" and it makes me wonder how this would look if we moved all of this into a subpackage of p2p called "gossipsub". I haven't digested enough of this yet to know if that has difficulties with the import graph / package-level symbol visibility.